submissions/madmaxWMFU/dom-task/js/app.js

```diff
@@ -0,0 +1,131 @@
+const menuItem = document.querySelector(".menu-wrap-list-item");
+const main = document.querySelector(".main-content");
+const numbersOfHeroes = 1491;
+
+menuItem.addEventListener("click", e => {
+    document.querySelector(".active").classList.remove("active");
+    e.target.parentElement.classList.add("active");
+    switch (e.target.dataset.href) {
+        case "home":
+            removeItems();
+            showLoad();
+            drawMainPage();
+            hideLoad();
+            break;
+        case "heros":
+            removeItems();
+            showLoad();
+            getDataApi(getRandomArray('characters'))
+                .then(json => {
+                    hideLoad();
+                    drawItems(json, "comiclink", "Comics zone");
+                })
+            break;
+        case "comics":
+            removeItems();
+            showLoad();
+            getDataApi(getRandomArray('comics'))
+                .then(json => {
+                    hideLoad();
+                    drawItems(json, "detail", "Detail");
+                });
+            break;
+    }
+});
+
+const getRandomArray = type => {
+    let arrayLinks = [];
+    while (arrayLinks.length < 10) {
+        let number = Math.floor((Math.random() * numbersOfHeroes) + 1);
+        let strLink = "";
+        if (arrayLinks.indexOf(number) === -1)
+            strLink = `https://gateway.marvel.com/v1/public/${type}?ts=1&limit=1&offset=${number}&apikey=0cc0bbc55224cb8f34ec109ab692c2cb&hash=4e7f342316da627d9a966768f39d0702`;
+        arrayLinks.push(strLink);
+    }
+    return arrayLinks;
+}
+
+const getDataApi = arrLinks => {
+    return new Promise(resolve => {
+        Promise.all(arrLinks.map(url => fetch(url).then(resp => resp.json())))
+            .then(result => resolve(result))
+            .catch(() => {
+                console.log('oops');
```

Comment:
Do you really need `console.log` in production?

Reply:
My simple mistake and inattention!(
All done!